contrib/storage-jdbc/src/main/java/org/apache/drill/exec/store/jdbc/JdbcStoragePlugin.java

```diff
@@ -291,27 +292,41 @@ public Table getTable(String name) {
       if (table != null) {
         return table;
       }
-      return inner.getTable(name.toUpperCase());
+      if (!areTableNamesCaseSensitive()) {
```

Comment:
Do we know which db we are using in the plugin? Maybe we can do table search via switch by db name rather than via if's with comments above?